kits/rosie/rosie_demo.cpp

```diff
@@ -0,0 +1,513 @@
+  
+    auto arm_traj = hebi::trajectory::Trajectory::createUnconstrainedQp(time, position);
+
+    auto t0 = fbk.getTime();
+    auto t = 0.0; 
+  
+    while(t < arm_min_traj_duration){
+      //
+      if(!robot_group->getNextFeedback(fbk)) printf("no feedback recieved\n");
+      auto temp_t = fbk.getTime();
+      t = (fbk.getTime() - t0) > arm_min_traj_duration ? arm_min_traj_duration : fbk.getTime() - t0;
+
+      getTrajectoryCommand(arm_traj, cmd, arm_dofs, t);
+  
+      //         TODO:   dynamicsComp
+      Eigen::VectorXd masses;
+      arm_model->getMasses(masses);
+      Eigen::VectorXd effort(arm_dofs.size());
+      auto base_accel = fbk[base_num_wheels].imu().accelerometer().get();
+      Eigen::Vector3d gravity(-base_accel.getX(),
+                     -base_accel.getY(),
+                     -base_accel.getZ());
+      Eigen::VectorXd arm_positions(arm_dofs.size());
+      for(size_t i = 0; i < arm_positions.size(); ++i){
+        arm_positions[i] = fbk.getPosition()[arm_dofs[i]];
+      }
+      effort = hebi::util::getGravityCompensationEfforts(*arm_model, masses, arm_positions, gravity);
+      /*TODO add dynamic_comp*/
+      setEffortSegment(cmd, arm_dofs, effort + arm_effort_offset);
+      robot_group->sendCommand(cmd);
+    }
+    //        % Grab initial pose
+    GroupFeedback latest_phone_mobile(phone_group->size());
+    while(!phone_group->getNextFeedback(latest_phone_mobile)){
+      printf("feedback not recieved\n\n");
+    }
+  
+    auto orient = latest_phone_fbk[0].mobile().arOrientation().get();
+    Eigen::Quaterniond q;
+    q.w() = orient.getW();
+    q.x() = orient.getX();
+    q.y() = orient.getY();
+    q.z() = orient.getZ();
+
+    auto r_init = q.toRotationMatrix(); 
+
+    auto ar_pos = latest_phone_fbk[0].mobile().arPosition().get();
+    Eigen::Vector3d xyz_init;
+    xyz_init << ar_pos.getX(), ar_pos.getY(), ar_pos.getZ();
+    auto xyz_phone_new = xyz_init;
+
+    Eigen::VectorXd end_velocities = Eigen::VectorXd::Zero(arm_dofs.size()); 
+    Eigen::VectorXd end_accels = Eigen::VectorXd::Zero(arm_dofs.size()); 
+
+    t0 = fbk.getTime();
+
+    auto time_last = t0;
+
+    auto chassis_traj_start_time = t0;  
+
+    GroupCommand wheel_cmd(wheel_group->size());
+    //
+    //        % Replan Trajectory for the mobile base
+    Eigen::Vector2d omni_base_traj_time;
+    omni_base_traj_time << 0, base_ramp_time;
+    //
+    //        % Initialize trajectory for Omnibase
+    Eigen::MatrixXd positions(3,2); 
+    Eigen::MatrixXd velocities = Eigen::MatrixXd::Zero(3,2); 
+    Eigen::MatrixXd accelerations = Eigen::MatrixXd::Zero(3,2); 
+    Eigen::MatrixXd jerks = Eigen::MatrixXd::Zero(3,2); 
+    
+    auto omni_base_traj = hebi::trajectory::Trajectory::createUnconstrainedQp(time, velocities, &accelerations, &jerks);
+    bool first_run = true;
+    
+    while(true){
+      //            %%%%%%%%%%%%%%%%%%%
+      //            % Gather Feedback %
+      //            %%%%%%%%%%%%%%%%%%%
```

Comment:
Note for future tidy/cleanup -- these are definitely MATLAB comments that were commented out in C++ :)   We can make this look more standard by just having these be a C++ comment characters, possibly with left alignment